File: Self_Solving_Algorithm/Cube_Algorithms/Tperm.py
from ..Cube_Turning.Cube_Turning_CCW import CubeArrayRCCW, CubeArrayUCCW
from ..Cube_Turning.Cube_Turning_CW import CubeArrayFCW, CubeArrayRCW
from ..Cube_Algorithms.Common_Moves.Sexy_Move import SexyMoveAlgorithm
from ..Cube_Algorithms.Common_Moves.Sledge_Hammer import SledgeHammerAlgorithm
import copy

Tperm_notation = [
    SexyMoveAlgorithm,
    CubeArrayRCCW,
    CubeArrayFCW,
    CubeArrayRCW,
    CubeArrayRCW,
    CubeArrayUCCW,
    CubeArrayRCCW,
    CubeArrayUCCW,
    SledgeHammerAlgorithm
]

def Tperm_algorithm(cube):
    current_state = copy.deepcopy(cube)
    for move in Tperm_notation:
        current_state = move(current_state)
    return current_state

File: Algorithm/cubeAlgorithms/Tperm.js
// (R U R' U') R' F R2 U' R' U' (R U R' F')

import { CubeArrayRCCW, CubeArrayUCCW } from "../cubeTurningCCW";
import { CubeArrayFCW, CubeArrayRCW } from "../cubeTurningCW";
import { SexyMoveAlgorithm } from "./commonMoves/sexyMove";
import { SledgeHammerAlgorithm } from "./commonMoves/sledgeHammer";

var TpermNotaiton = [
  SexyMoveAlgorithm,
  CubeArrayRCCW,
  CubeArrayFCW,
  CubeArrayRCW,
  CubeArrayRCW,
  CubeArrayUCCW,
  CubeArrayRCCW,
  CubeArrayUCCW,
  SledgeHammerAlgorithm,
];

export function TpermAlgorithm(cube) {
  let CurrentState = JSON.parse(JSON.stringify(cube));
  for (let i = 0; i < TpermNotaiton.length; i++) {
    CurrentState = TpermNotaiton[i](CurrentState);
  }

  return CurrentState;
}
